### PR TITLE
refactor sgrep py

### PR DESCRIPTION
**Pull request by ievans**
What's this? Separate files for the python core!

Also, Docker now builds sgrep.py into a **single, portable** binary named sgrep-lint 

The dockerfile is also almost half the size:

```
sgrep-refactored                          53.9MB 
returntocorp/sgrep:develop       82.8MB
```

And it's significantly faster (note that the majority of the runtime here is sgrep core, so the savings are significant):
```
ine@imbp4 ~/D/r/sgrep-rules (feature/prototype-go-rules)> hyperfine -m 20 --warmup 2 'docker run --rm -v $(pwd):/home/repo sgrep-local:latest --config=. python/'
Benchmark #1: docker run --rm -v $(pwd):/home/repo sgrep-local:latest --config=. python/
  Time (mean ± σ):     **10.426 s ±  0.312 s**    [User: 26.8 ms, System: 13.5 ms]
  Range (min … max):    9.857 s … 10.960 s    20 runs

ine@imbp4 ~/D/r/sgrep-rules (feature/prototype-go-rules)> hyperfine -m 20 --warmup 2 'docker run --rm -v $(pwd):/home/repo returntocorp/sgrep:develop --config=. python/'
Benchmark #1: docker run --rm -v $(pwd):/home/repo returntocorp/sgrep:develop --config=. python/
  Time (mean ± σ):     **10.814 s ±  0.241 s**    [User: 26.9 ms, System: 13.5 ms]
  Range (min … max):   10.277 s … 11.351 s    20 runs
```

Followons:
* sgrep -> sgrep-core, sgrep-lint -> sgrep
* move test.py functionality into sgrep.py

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,14 +3,13 @@ version: 2.1
 executors:
   sgrep-build:
     docker:
-      - image: returntocorp/sgrep-build:2.5
+      - image: returntocorp/sgrep-build:latest
```
Comment (DrewDennison) [inline .circleci/config.yml]: this should be a tag so Circle can cache it but I want to replace this with a GH action next week as part of the mac build so ok for now

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -1,29 +0,0 @@
...
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
-    - name: Test with pytest
-      run: |
-        # add to PYTHONPATH so imports work in test files
-        export PYTHONPATH="${PYTHONPATH}:${pwd}"
-        pip install pytest
-        pytest
```
Comment (DrewDennison) [inline .github/workflows/python.yml]: I assume you use something other than pytest so ok with deleting

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,12 +1,11 @@
-exclude: "^tests|testlint|old|lib|demos|bin|.*\\.ml"
+exclude: "^sgrep-lint/tests|tests|testlint|old|lib|demos|bin|.*\\.ml"
 
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v2.5.0
     hooks:
       - id: check-yaml
         args: [--allow-multiple-documents]
-      - id: end-of-file-fixer
```
Comment (DrewDennison) [inline .pre-commit-config.yaml]: why? 😞

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -26,17 +25,6 @@ repos:
     hooks:
       - id: reorder-python-imports
 
-  - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: "v0.761"
-    hooks:
-      - id: mypy
```
Comment (DrewDennison) [inline .pre-commit-config.yaml]: I thought we believed in types 😢 Would you be ok if I added in the types to keep this check

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -26,17 +25,6 @@ repos:
...
-
-  - repo: https://gitlab.com/pycqa/flake8
-    rev: "3.7.9"
-    hooks:
-      - id: flake8
-        args: ["--select=E9,F63,F7,F82"]
```
Comment (DrewDennison) [inline .pre-commit-config.yaml]: I have used this config multiple times to catch bugs

File: .pre-commit-config.yaml (hunk 3 above)
Comment (ievans) [reply .pre-commit-config.yaml]: breaks test cases

File: .pre-commit-config.yaml (hunk 4 above)
Comment (ievans) [reply .pre-commit-config.yaml]: mypy doesn't handle the weird globals and eval nonsense

File: .pre-commit-config.yaml (hunk 5 above)
Comment (ievans) [reply .pre-commit-config.yaml]: is it doing anything that mypy wouldn't? we have tons of test files with intentionally unused variables

File: .github/workflows/python.yml (hunk 2 above)
Comment (ievans) [reply .github/workflows/python.yml]: It's just part of `make test` now

File: sgrep_lint/constants.py [hunk 6]
```diff
@@ -0,0 +1 @@
+RCE_RULE_FLAG = "--dangerously-allow-arbitrary-code-execution-from-rules"
```
Comment (DrewDennison) [inline sgrep_lint/constants.py]: 🗡

**Pull request merged**